### DISCUSSION TITLE
How to use same decor for controller

### DISCUSSION THREAD

**Discussion by pankajjadhav-gep** [lawrence-laz/Decor.NET#14] [Q&A]
Hi @lawrence-laz ,

This package is really helpful. I have one question.

can I use same decor on method which is available in my controller ?

 How to use? please share any codebase or example

Thanks in advance

**Comment by lawrence-laz** [comment]
Hi @pankajjadhav-gep ,

Glad you find the package helpful.

Using decorators on asp.net controllers is difficult. I don't know if there even is a right way of doing it. That is due to [how asp.net controllers are being constructed](https://source.dot.net/#Microsoft.AspNetCore.Mvc.Core/Controllers/ControllerActivatorProvider.cs,6c95364515655c93,references), which makes it difficult to attach dynamic proxy interceptors.

This might change in the future for Decor version 3, which will be using source generators instead of dynamic proxies. But no guarantees.

For now, I would suggest to go with asp.net's way of intercepting requests on controller level either using [a middleware](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/middleware/?view=aspnetcore-5.0) or [a filter](https://docs.microsoft.com/en-us/aspnet/core/mvc/controllers/filters?view=aspnetcore-5.0).

Sorry for not being of much help.

**Reply by pankajjadhav-gep** [reply]
Thanks @lawrence-laz for reply.
Will try your suggested solutions. But yes will be great if we get in future Decor on Public methods.
